force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -101,18 +107,21 @@ public with sharing class ClassesObjectsMethods {
      * @return The number of vowels in the string.
      */
     public static Integer countVowels(String str) {
-        // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        // Convert the string to lowercase for case insensitivity
+        str = str.toLowerCase();
+        // Initialize the count of vowels
+        Integer count = 0;
+        // Loop through each character in the string
+        for (Integer i = 0; i < str.length(); i++) {
+            // Get the current character
+            String ch = str.substring(i, i + 1);
+            // Check if the character is a vowel
+            if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') {
+                count++;
+            }
```

Comment:
I really like the variable names and choices here, but for your if statement, you can check if a Set<String> or String of all the vowels contains the current character. This is solid logic though for a programming language where we might not have libraries for these things.